src/Engine/Data/EnvironmentTexture.cpp

```diff
@@ -0,0 +1,739 @@
+#include <algorithm>
+#include <cmath>
+#include <filesystem>
+#include <iostream>
+
+#include <cstring>
+
+#include <Engine/Data/EnvironmentTexture.hpp>
+
+//#define STB_IMAGE_IMPLEMENTATION
+#include <stb/stb_image.h>
+//#undef STB_IMAGE_IMPLEMENTATION
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+#include <stb/stb_image_write.h>
+#undef STB_IMAGE_WRITE_IMPLEMENTATION
+
+#include <Core/Asset/Camera.hpp>
+#include <Core/Geometry/MeshPrimitives.hpp>
+#include <Core/Math/Math.hpp>
+#include <Core/Resources/Resources.hpp>
+
+#include <Engine/Data/Mesh.hpp>
+#include <Engine/Data/ShaderProgram.hpp>
+#include <Engine/Data/ShaderProgramManager.hpp>
+#include <Engine/Data/Texture.hpp>
+#include <Engine/Data/ViewingParameters.hpp>
+#include <Engine/RadiumEngine.hpp>
+#include <Engine/Scene/CameraManager.hpp>
+
+#include <tinyEXR/tinyexr.h>
+
+namespace Ra {
+namespace Engine {
+namespace Data {
+using namespace gl;
+
+// Flip horizontally an image of w x h pixels with c commponents
+template <typename T>
+void flip_horizontally( T* img, size_t w, size_t h, size_t c ) {
+#pragma omp parallel for
```

Comment:
```suggestion
#pragma omp parallel for collapse(2)
```

Reply:
This clause generates segfault on std::swap on my test machine (llvm Clang 11.1.0, openmp 5.0)